payjoin/src/receive/mod.rs

```diff
@@ -229,14 +228,8 @@ impl MaybeMixedInputScripts {
         let input_scripts = self
             .psbt
             .input_pairs()
-            .scan(&mut err, |err, input| match input.previous_txout() {
-                Ok(txout) => match InputType::from_spent_input(txout, input.psbtin) {
-                    Ok(input_script) => Some(input_script),
-                    Err(e) => {
-                        **err = Err(RequestError::from(InternalRequestError::InputType(e)));
-                        None
-                    }
-                },
+            .scan(&mut err, |err, input| match Ok(input.address_type()) {
+                Ok(address_type) => Some(address_type),
```

Comment:
Seems like this will never reach the `Err(e)` branch since match is directly called on an `Ok` value?

Reply:
Yes - this was a temporary workaround before I had `address_type()` return a `Result`. It gets updated in https://github.com/payjoin/rust-payjoin/pull/360/commits/6938b70f486a3999063fb2218766212ee5855c99#diff-5fba6fc39c59b07e5980d87da92537138068364d192a2b63eee5e57af01a07bb